lti_tool/views.py

```diff
@@ -0,0 +1,320 @@
+from django.http import HttpResponse
+from django.views.decorators.clickjacking import xframe_options_exempt
+from django.views.decorators.csrf import csrf_exempt
+from django.shortcuts import redirect
+from django.contrib.auth.models import User
+from django.contrib.auth import login
+from django.urls import reverse
+from pylti1p3.contrib.django import DjangoOIDCLogin, DjangoMessageLaunch
+from pylti1p3.deep_link_resource import DeepLinkResource
+from pylti1p3.lineitem import LineItem
+from pylti1p3.exception import LtiException
+
+from course.views import InstanceView, ModuleView
+from course.models import CourseInstance, Enrollment, CourseModule
+from course.viewbase import CourseInstanceBaseView, CourseModuleBaseView
+from exercise.viewbase import ExerciseBaseView
+from exercise.models import LearningObject, BaseExercise
+from exercise.views import ExerciseView, SubmissionView
+from userprofile.models import UserProfile
+from lib.viewbase import BaseTemplateView, BaseRedirectView, BaseMixin
+from authorization.permissions import ACCESS
+from .utils import get_tool_conf, get_launch_data_storage, get_launch_url
+
+# CSRF exempts due to difficulty in proper implementation. To re-review later.
+@csrf_exempt
+@xframe_options_exempt
+def lti_login(request):
+    tool_conf = get_tool_conf()
+    launch_data_storage = get_launch_data_storage()
+
+    oidc_login = DjangoOIDCLogin(request, tool_conf, launch_data_storage=launch_data_storage)
+    target_link_uri = get_launch_url(request)
+    return oidc_login.enable_check_cookies().redirect(target_link_uri)
+
+
+class LtiLaunchView(BaseRedirectView):
+
+    access_mode = ACCESS.ANONYMOUS
+
+    @csrf_exempt
+    @xframe_options_exempt
+    def post(self, request, *args, **kwargs):
+        tool_conf = get_tool_conf()
+        launch_data_storage = get_launch_data_storage()
+        message_launch = DjangoMessageLaunch(self.request, tool_conf, launch_data_storage=launch_data_storage)
+        self.message_launch_data = message_launch.get_launch_data()
+        
+        # Get or create user
+        username = self.message_launch_data["email"]
+        try:
+            # Moodle sends username in this parameter
+            username = self.message_launch_data['https://purl.imsglobal.org/spec/lti/claim/ext']['user_username']
+        except (KeyError, TypeError):
+            # Email always received, use as fallback
+            username = self.message_launch_data['email']
+        users = User.objects.filter(username=username)
```

Comment:
`username` is unique. You should use `User.objects.get()` and catch the exception when the user does not exist.